rotkehlchen/db/utils.py

```diff
@@ -363,3 +372,84 @@ def combine_asset_balances(balances: List[SingleDBAssetBalance]) -> List[SingleD
             new_balances = _append_or_combine(new_balances, balance)
 
     return new_balances
+
+
+def db_schema_sanity_check(
+        db_name: Literal['encrypted', 'global'],
+        cursor: 'DBCursor',
+        script_create_tables: str,
+) -> None:
+    """Ensures that database schema is not broken. Raises DBSchemaError if anything is bad."""
+    # Replacing new lines and white spaces since they may vary if by an accident code of a
+    # db upgrade was a bit different from the one that creates new tables
+    create_script_from_constant = script_create_tables.replace(' ', '').replace('\n', '').replace('\'', '"')  # noqa: E501
+
+    # Make dictionary of table_name: table_properites. Without whitespaces. For example:
+    #   "ens_mappings": "CREATETABLEIFNOTEXISTSens_mappings(addressTEXTNOTNULLPRIMARYKEY,ens_nameTEXTUNIQUE,last_updateINTEGERNOTNULL);"  # noqa: E501
+    #
+    tables_data_from_constant = {name: properties for name, properties in re.findall(
+        pattern=r'CREATETABLEIFNOTEXISTS(.+?)\((.+?)\);',
+        string=create_script_from_constant,
+    )}
+
+    # Create same mapping for tables that are currently in the database
+    cursor.execute('SELECT name, sql FROM sqlite_master WHERE type="table"')
+    tables_data_from_db = {}
+    for (name, raw_script) in cursor:
+        minimized_script = raw_script.replace(' ', '').replace('\n', '').replace('\'', '"')
+        table_properties = re.findall(
+            pattern=r'CREATETABLE.*?\((.+)\)',
+            string=minimized_script,
+        )[0]
+        # Store table properties for comparison with expected table structure and raw_script
+        # for the ease of debugging
+        tables_data_from_db[name] = (table_properties, raw_script)
+
+    # Check that there are no extra structures such as views
+    extra_db_sctructures = cursor.execute(
+        'SELECT * FROM sqlite_master WHERE type NOT IN ("table", "index")',
+    ).fetchall()
+    if len(extra_db_sctructures) > 0:
```

Comment:
Sorry, I did not really understand the concern and the proposal.

Could you give an example of how it can hit?

will `f'{structure_type}{table_name}' not in create_script_from_constant` work? I don't think so. I think it will never be true

Reply:
He means if ever sqlite_maste table adds another column. And he is right. Check the 2 specific columns, not `*`

About the 2nd part ofthe suggestion I am not sure I get what @yabirgb meant

Reply:
We discussed and @nebolax convinced me that is the better approach. The issue with this is that it if we ever add a new view for example the app will break but until we fix this test. In my mind it was bad because it means touching code unrelated to the topic and having to create all the logic to for example handling views adapting this but is the best approach since it will warn as instead of ignoring

Reply:
Okay but the first comment about not using `*` is correct and should be applied.